website/static/js/fangorn.js

```diff
@@ -532,8 +546,22 @@ function doItemOp(operation, to, from, rename, conflict) {
  * @private
  */
 function _fangornResolveUploadUrl(item, file) {
+    // WB v1 update synatx is file path and kind=file
+    // WB v1 upload synatx is parent path and kind=file, name=filename
+    // If upload target file name already exists don't pass file.name
+    var updateUrl;
     var configOption = resolveconfigOption.call(this, item, 'uploadUrl', [item, file]); // jshint ignore:line
-    return configOption || waterbutler.buildTreeBeardUpload(item, file);
+    if (configOption) {
+        return configOption;
+    } else {
```

Comment:
Minor: you don't need to wrap the rest of the code in an `else` here if the true condition returns.  You can also move the declaration of `updateUrl` down closer to it's assignment.